### PR TITLE
Fix Network

### PR DESCRIPTION
**Pull request by onmyway133**
- Remove Sugar and Alamofire
- Use URLSession for network request
- Make Networking static variable. Allow configuration and subclass

### CODE REVIEW

**Comment by vadymmarkov** [comment]
@onmyway133 My question is...why not Malibu?

**Comment by onmyway133** [comment]
@vadymmarkov you know ... 😄  But this is a small post request, so I think it helps to easily track down what went wrong in the future

File: Sources/Shared/Networking/NetworkTasks.swift [hunk 1]
```diff
@@ -54,7 +53,7 @@ struct TokenNetworkTask: NetworkTaskable, NetworkQueueable {
       } catch {}
     }
 
-    locker.synchronize()
+    let _ = locker.synchronize()
```
Comment (vadymmarkov) [inline Sources/Shared/Networking/NetworkTasks.swift]: I'd better use @discardableResult

File: Sources/Shared/Networking/Networking.swift [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import Foundation
+
+open class Networking {
```
Comment (vadymmarkov) [inline Sources/Shared/Networking/Networking.swift]: @onmyway133 Should it be open? I think we can go with internal here.

File: Sources/Shared/Networking/Networking.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  public let session: URLSession
+  
+  public init(configuration: URLSessionConfiguration) {
+    self.session = URLSession(configuration: configuration)
```
Comment (vadymmarkov) [inline Sources/Shared/Networking/Networking.swift]: `self` could be removed 😄

Comment (onmyway133) [inline Sources/Shared/Networking/Networking.swift]: Yeah

File: Sources/Shared/Networking/Networking.swift (hunk 2 above)
Comment (onmyway133) [inline Sources/Shared/Networking/Networking.swift]: @vadymmarkov I think just in case someone wants to override the `post` method

Comment (vadymmarkov) [inline Sources/Shared/Networking/Networking.swift]: Yeah, but we can add if needed I think. I'd better simplify the public API.

Comment (onmyway133) [inline Sources/Shared/Networking/Networking.swift]: @vadymmarkov User can configure the `Networking` by assigning new one to the `AuthConfig`, so it should be either `public` or `open`

Comment (vadymmarkov) [inline Sources/Shared/Networking/Networking.swift]: Ok then 😄

**Comment by vadymmarkov** [comment]
@onmyway133 Great work!

**Pull request merged**